### PR TITLE
refactor(onClickOutside): better safari workaround

### PR DESCRIPTION
**Pull request by sibbng**
### Description

Better fix for #1520. I didn't read the article I mentioned in #1522 carefully. It seems there is a better fix for it. 

Also, it seems this issue is fixed in the latest version of Safari.

### Additional context

---

### What is the purpose of this pull request? 

- [ ] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [x] Other

### Before submitting the PR, please make sure you do the following

- [ ] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [ ] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [ ] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/core/onClickOutside/index.ts [hunk 1]
```diff
@@ -66,18 +62,13 @@ export function onClickOutside<T extends OnClickOutsideOptions>(
   }
 
   const cleanup = [
+    [].slice.call(window.document.body.children)
+      .map(el => useEventListener(el, 'click', () => {})),
```
Comment (antfu) [inline packages/core/onClickOutside/index.ts:66]: This feels too expensive and I am not sure about it.

Comment (sibbng) [reply packages/core/onClickOutside/index.ts:66]: It's not. But we could optimize it. Currently, these event handlers will be attached on each use of `onClickOutside`. We could make it a one-time operation, initialize it only on iOS, and filter out `<script>`, `<link>` and `<style>` elements.

**Comment by sibbng** [comment]
A quick note about why I use `addEventListener` instead of `useEventListener`. I don't want these listeners to be removed when scope disposed as they are attached only on the first use of `onClickOutside`. Since we don't need to remove listeners on scope disposal relying on `addEventListener` removes the overhead of `useEventListener` and makes the memory footprint smaller.

File: packages/core/onClickOutside/index.ts [hunk 2]
```diff
@@ -43,8 +45,8 @@ export function onClickOutside<T extends OnClickOutsideOptions>(
   if (!window)
     return
 
-  if (!(window as any).__VUEUSE_ONCLICKOUTSIDE__ && /iPad|iPhone|iPod/.test(navigator.userAgent)) {
-    (window as any).__VUEUSE_ONCLICKOUTSIDE__ = true
+  if (!_iOSWorkaround && isIOS) {
```
Comment (antfu) [inline packages/core/onClickOutside/index.ts:48]:
```suggestion
  if (isIOS && !_iOSWorkaround) {
```

**Review by antfu — APPROVED** [review]

File: packages/core/onClickOutside/index.ts [hunk 3]
```diff
@@ -42,9 +45,13 @@ export function onClickOutside<T extends OnClickOutsideOptions>(
   if (!window)
     return
 
-  let shouldListen = true
+  if (isIOS && !_iOSWorkaround) {
+    _iOSWorkaround = true
+    Array.from(window.document.body.children)
```
Comment (antfu) [inline packages/core/onClickOutside/index.ts:50]: Are you able to provide some reference to explain why this is needed in the comments?

**Pull request merged**